cmd/cloudinit/cloudinit.go

```diff
@@ -229,7 +230,9 @@ func saveCloudConfig() error {
 		}
 	}
 
-	userData := string(userDataBytes)
+	env := initialize.NewEnvironment("", "", "", "", metadata)
+	userData := env.Apply(string(userDataBytes))
+	userDataBytes = []byte(userData)
```

Comment:
@imikushin to be clear, you want me to implement them as a function in the unit test file, or you also want them as a function in the cloudinit.go file?